cmd/juju/charmhub/data.go

```diff
@@ -111,9 +111,36 @@ func convertChannels(in map[string]charmhub.Channel, series string) map[string]C
 			Version:    v.Version,
 		}
 	}
+
 	return out
 }
 
+func filterChannels(in map[string]charmhub.Channel, architecture, series string) map[string]charmhub.Channel {
+	allArch := architecture == ArchAll
+	allSeries := series == SeriesAll
+
+	// If we're searching for everything then we can skip the filtering logic
+	// and return immediately.
+	if allArch && allSeries {
+		return in
+	}
+
+	// Channels that match any part of the criteria should be witnessed and
+	// kept.
+	witnessed := make(map[string]charmhub.Channel)
+
+	for k, v := range in {
+		archSet := channelArches(v.Platforms)
+		seriesSet := channelSeries(v.Platforms)
+
+		if (allArch || archSet.Contains(architecture) || archSet.Contains(ArchAll)) &&
```

Comment:
Does the charmhub API return the literal "all" or is that a mapping applied by our client?

Reply:
Yeah, it literally does ;-)